nemo/core/neural_factory.py

```diff
@@ -154,29 +344,29 @@ def infer(self, tensors: List[NmTensor]):
         """
         pass
 
-    @abstractmethod
-    def save_state_to(self, path: str):
-        """
-        Saves current state such as step, epoch and optimizer parameters
-        Args:
-          path:
+    # @abstractmethod
+    # def save_state_to(self, path: str):
+    #     """
+    #     Saves current state such as step, epoch and optimizer parameters
+    #     Args:
+    #       path:
```

Comment:
This seems to be an ingerence in NF API. Following our official guidelines - shoudn't we make it @deprecated first?

Reply:
I'd say that exposing save_state_to to the user was a mistake in the first place